irmagobridge/action_handler.go

```diff
@@ -216,3 +219,55 @@ func (ah *ActionHandler) updateSchemes() error {
 	sendConfiguration()
 	return nil
 }
+
+type LoadLogsAction struct {
+	Before string
+	Max int
+}
+
+func (ah *ActionHandler) LoadLogs(action *LoadLogsAction) error {
+	beforeInt, err := strconv.ParseInt(action.Before, 10, 64)
+	if err != nil {
+		return err
+	}
+	logEntries, err := client.LoadLogs(time.Unix(beforeInt, 0), action.Max)
+	if err != nil {
+		logError(errors.WrapPrefix(err, "Could not collect logs to send", 0))
+		return err
+	}
+
+	logsOutgoing := make([]*OutgoingAction, len(logEntries))
+	for i, entry := range logEntries {
+		var removed map[irma.CredentialTypeIdentifier][]irma.TranslatedString
+		if entry.Type == irmaclient.ActionRemoval {
+			removed = entry.Removed
+		} else {
+		    removed = make(map[irma.CredentialTypeIdentifier][]irma.TranslatedString)
+		}
+		disclosedCredentials, err := entry.GetDisclosedCredentials(client.Configuration)
+		if err != nil {
+			return err
+		}
+		issuedCredentials, err := entry.GetIssuedCredentials(client.Configuration)
+		if err != nil {
+			return err
+		}
+		signedMessage, err := entry.GetSignedMessage()
+		if err != nil {
+			return err
+		}
+		entry.SessionRequest()
+		logsOutgoing[i] = &OutgoingAction{
```

Comment:
Looking at the rest of the code, the purpose of `OutgoingAction` seems to be to serve as argument to `sendAction()`. It's better to just use `map[string]interface{}` here (or create a new struct).

Reply:
This was how it was done in the code I got from @confiks, but another data structure is maybe more clear indeed.